pkg/proxy/proxy.go

```diff
@@ -149,6 +168,78 @@ func (p *Proxy) StartProxy() *http.Server {
 	return srv
 }
 
+// unsecured returns true if the request does not require authentication
+func unsecured(ctx echo.Context) bool {
+	uri := ctx.Request().URL.RequestURI()
+	return uri == proxyHealthEndpoint || uri == "/.well-known/oauth-authorization-server" || strings.HasPrefix(uri, authEndpoint)
+}
+
+// auth handles requests to SSO. Used by web login.
+func (p *Proxy) auth(ctx echo.Context) error {
+	req := ctx.Request()
+	targetURL, err := url.Parse(configuration.GetRegistrationServiceConfig().Auth().SSOBaseURL())
+	if err != nil {
+		return err
+	}
+	targetURL.Path = req.URL.Path
+	targetURL.RawQuery = req.URL.RawQuery
+
+	return p.handleSSORequest(targetURL)(ctx)
+}
+
+// oauthConfiguration handles requests to oauth configuration and proxies them to the corresponding SSO endpoint. Used by web login.
+func (p *Proxy) oauthConfiguration(ctx echo.Context) error {
+	targetURL, err := url.Parse(ssoWellKnownTarget())
+	if err != nil {
+		return err
+	}
+	return p.handleSSORequest(targetURL)(ctx)
+}
+
+// openidAuth handles requests to the openID Connect authentication endpoint. Used by web login.
+func (p *Proxy) openidAuth(ctx echo.Context) error {
+	targetURL, err := url.Parse(authorizationEndpointTarget())
+	if err != nil {
+		return err
+	}
+	targetURL.Path = ctx.Request().URL.Path
+	targetURL.RawQuery = ctx.Request().URL.RawQuery
+
+	return p.redirectTo(ctx, targetURL.String())
```

Comment:
We need to use redirect instead of `handleSSORequest` because this is technically the web login in the browser, is that correct?

Reply:
See https://github.com/codeready-toolchain/registration-service/pull/382/commits/5a8e586828d7e638e5cc85c94d28e8dc1e25302c

Reply:
gotcha, thanks for the explanation.